Create new data loader page (#319)
HarperDB/documentation: versioned_docs/version-4.6/developers/applications/loading-data.md

> ---

Reviewer: ...that's not a great workflow though and we _could_ do something with syncing changes from 4.6 when 4.7 is cut or something. probably a low pri, if anything, but up to you and Ethan.

> ---

Reviewer 2: yeah we are just copy/pasting right now for relevant versions. Its what it is for now. We have a lot to due before 10/14 and not really time to worry about process or automation. Few extra clicks here and there is no issue.

First, let’s make a `data` directory in our app and create a file called `dogs.json`:

```json
{
	"database": "myapp",
	"table": "Dog",
	"records": [
		{
			"id": 1,
			"name": "Harper",
			"breed": "Labrador",
			"age": 3,
			"tricks": ["sit"]
		},
		{
			"id": 2,
			"name": "Balto",
			"breed": "Husky",
			"age": 5,
			"tricks": ["run", "pull sled"]
		}
	]
}
```

This file tells Harper: _“Insert these two records into the `Dog` table when this app runs.”_

## Connecting the Data Loader

Next, let’s tell Harper to use this file when running the application. Open `config.yaml` in the root of your project and add:

```yaml
dataLoader:
  files: 'data/dogs.json'
```

That’s it. Now the Data Loader knows where to look.

## Running with Data

Go ahead and start your app again:

```bash
harperdb dev .
```

This time, when Harper runs, it will automatically read `dogs.json` and load the records into the Dog table. You don’t need to write any import scripts or SQL statements, it just works.

You can confirm the data is there by hitting the endpoint you created earlier:

```bash
curl http://localhost:9926/Dog/
```

You should see both `Harper` and `Balto` returned as JSON.

:::info

> 💡 Notice the trailing `/` in the URL (`/Dog/`). This tells Harper to return all records in the table. Leaving it off would look for a single record instead.

Reviewer 2: ```suggestion
💡 Notice the trailing `/` in the URL (`/Dog/`). This tells Harper to return all records in the table. Leaving it off would result in a description of the `Dog` table instead.
```
